cardano-testnet/src/Testnet/LedgerEventProcessing.hs

```diff
@@ -0,0 +1,46 @@
+module Testnet.LedgerEventProcessing
+  ( filterNewGovProposals
+  , foldBlocksFindLedgerEvent
+  ) where
+
+
+import           Cardano.Api as Api
+import           Cardano.Api.Shelley (AnyProposals (AnyProposals),
+                   LedgerEvent (NewGovernanceProposals), fromShelleyTxId)
+
+import qualified Cardano.Ledger.Conway.Governance as Ledger
+
+import           Prelude
+
+import           Control.Monad.Catch (MonadCatch)
+
+foldBlocksFindLedgerEvent :: (MonadIO m, MonadCatch m)
+  => (LedgerEvent -> Bool)
+  -> FilePath
+  -> FilePath
+  -> m (Either FoldBlocksError (Maybe LedgerEvent))
+foldBlocksFindLedgerEvent ledgerEventFilter configurationFile socketPath =
+    runExceptT $ foldBlocks (File configurationFile)
```

Comment:
We should be using `foldEpochState`

Reply:
Addressed in https://github.com/IntersectMBO/cardano-node/pull/5731/commits/e386bb7f221f6c4c00c5ab3331acaa173c415b17